core/client/src/main/java/alluxio/client/block/RemoteBlockOutStream.java

```diff
@@ -53,8 +58,8 @@ public RemoteBlockOutStream(long blockId,
     super(blockId, blockSize, blockStoreContext);
     mCloser = Closer.create();
     try {
-      mRemoteWriter = mCloser.register(RemoteBlockWriter.Factory.create());
       mBlockWorkerClient = mCloser.register(mContext.createWorkerClient(address));
+      mRemoteWriter = mCloser.register(RemoteBlockWriter.Factory.create());
```

Comment:
why change the order here? is it by purpose?
if there is something tricky, we should put a comment here